server/ComputeSuperpixelFeatures/ComputeSuperpixelFeatures.py

```diff
@@ -0,0 +1,558 @@
+        #
+        tile_fgnd_frac_list = [1.0]
+
+        it_kwargs = {
+            'tile_size': {'width': args.analysis_tile_size},
+            'scale': {'magnification': args.analysis_mag},
+        }
+
+        if not process_whole_image:
+
+            it_kwargs['region'] = {
+                'left':   args.analysis_roi[0],
+                'top':    args.analysis_roi[1],
+                'width':  args.analysis_roi[2],
+                'height': args.analysis_roi[3],
+                'units':  'base_pixels'
+            }
+
+        if is_wsi:
+
+            print('\n>> Computing foreground fraction of all tiles ...\n')
+
+            start_time = time.time()
+
+            num_tiles = ts.getSingleTile(**it_kwargs)['iterator_range']['position']
+
+            print 'Number of tiles = %d' % num_tiles
+
+            tile_fgnd_frac_list = htk_utils.compute_tile_foreground_fraction(
+                img_paths[i], im_fgnd_mask_lres, fgnd_seg_scale,
+                **it_kwargs
+            )
+
+            num_fgnd_tiles = np.count_nonzero(
+                tile_fgnd_frac_list >= args.min_fgnd_frac)
+
+            percent_fgnd_tiles = 100.0 * num_fgnd_tiles / num_tiles
+
+            fgnd_frac_comp_time = time.time() - start_time
+
+            print 'Number of foreground tiles = %d (%.2f%%)' % (
+                num_fgnd_tiles, percent_fgnd_tiles)
+
+            print 'Time taken = %s' % cli_utils.disp_time_hms(fgnd_frac_comp_time)
+
+        #
+        # Detect superpixel data in parallel using Dask
+        #
+        print('\n>> Detecting superpixel data ...\n')
+
+        start_time = time.time()
+
+        tile_result_list = []
+
+        for tile in ts.tileIterator(**it_kwargs):
+
+            tile_position = tile['tile_position']['position']
+
+            if is_wsi and tile_fgnd_frac_list[tile_position] <= args.min_fgnd_frac:
+                continue
+
+            # detect superpixel data
+            cur_result = dask.delayed(compute_superpixel_data)(
+                img_paths[i],
+                tile_position,
+                wsi_mean, wsi_stddev,
+                args, **it_kwargs)
+
+            # append result to list
+            tile_result_list.append(cur_result)
+
+        tile_result_list = dask.delayed(tile_result_list).compute()
+
+        superpixel_data = []
+        x_centroids = []
+        y_centroids = []
+        x_boundaries = []
+        y_boundaries = []
+
+        for s_data, x_cent, y_cent, x_brs, y_brs in tile_result_list:
+
+            for s_d in s_data:
+                superpixel_data.append(s_d)
+
+            for x_c in x_cent:
+                x_centroids.append(x_c)
+
+            for y_c in y_cent:
+                y_centroids.append(y_c)
+
+            for x_b in x_brs:
+                x_boundaries.append(x_b)
+
+            for y_b in y_brs:
+                y_boundaries.append(y_b)
+
+        superpixel_data = np.asarray(superpixel_data, dtype=np.float32)
```

Comment:
This size of `superpixel_data` could be huge. It would essentially be equal to all foreground part of the image that can be quite large. The scheduler machine may not have enough RAM to store it.

I would suggest moving the autoencoder feature extraction part into the function `compute_super_pixel_data` that you parallelize with dask.